diff --git a/libavformat/http.c b/libavformat/http.c
--- a/libavformat/http.c
+++ b/libavformat/http.c
@@ -889,10 +889,8 @@ static int check_http_code(URLContext *h, int http_code, const char *end)
 {
-    HTTPContext *s = h->priv_data;
     /* error codes are 4xx and 5xx, but regard 401 as a success, so we
      * don't abort until all headers have been parsed. */
     if (http_code >= 400 && http_code < 600 &&
-        (http_code != 401 || s->auth_state.auth_type != HTTP_AUTH_NONE) &&
-        (http_code != 407 || s->proxy_auth_state.auth_type != HTTP_AUTH_NONE)) {
+        http_code != 401 && http_code != 407) {
         end += strspn(end, SPACE_CHARS);
         av_log(h, AV_LOG_WARNING, "HTTP error %d %s\n", http_code, end);
         return ff_http_averror(http_code, AVERROR(EIO));
